src/guiguts/tools/levenshtein.py

```diff
@@ -0,0 +1,683 @@
+"""Levenshtein edit distance check functionality"""
+
+import time
+import importlib.resources
+import logging
+
+from enum import StrEnum, auto
+from tkinter import ttk
+from Levenshtein import distance
+
+import regex as re
+
+from guiguts.checkers import CheckerDialog
+from guiguts.spell import SpellChecker
+from guiguts.data import dictionaries
+from guiguts.file import ProjectDict
+from guiguts.maintext import maintext
+from guiguts.misc_tools import tool_save
+from guiguts.preferences import (
+    PersistentString,
+    PrefKey,
+)
+from guiguts.utilities import IndexRowCol, IndexRange
+from guiguts.widgets import ToolTip
+
+logger = logging.getLogger(__package__)
+
+DEFAULT_DICTIONARY_DIR = importlib.resources.files(dictionaries)
+
+SPELL_CHECK_OK_YES = 0
+SPELL_CHECK_OK_NO = 1
+SPELL_CHECK_OK_BAD = 2
+
+LINES_IN_REPORT_LIMIT = 2
+
+_the_levenshtein_checker = None  # pylint: disable=invalid-name
+
+
+#########################################################
+# levenshtein.py
+# Python author: Quentin Campbell (DP:qgc) - 2024.
+# Based on the processing logic in the PPtxt Go version
+# authored by Roger Franks (DP:rfrank - 2020) but using
+# a publicly available Levenshtein module authored by
+# Max Bachmann.
+#########################################################
+
+
+class LevenshteinEditDistance(StrEnum):
+    """Enum class to store Levenshtein Edit Distance."""
+
+    ONE = auto()
+    TWO = auto()
+
+
+class DictionaryNotFoundError(Exception):
```

Comment:
Probably should import this exception from `guiguts.spell` rather than define a new class here - it is SpellChecker that ends up raising this.